hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/HandleThrottleAccumulator.java

```diff
@@ -0,0 +1,427 @@
+                new EnumMap<>(HederaFunctionality.class);
+
+        for (var bucket : defs.throttleBuckets()) {
+            try {
+                final var utilThrottleBucket = new ThrottleBucket<>(
+                        bucket.burstPeriodMs(),
+                        bucket.name(),
+                        bucket.throttleGroups().stream()
+                                .map(this::hapiGroupFromPbj)
+                                .toList());
+                var mapping = utilThrottleBucket.asThrottleMapping(CAPACITY_SPLIT);
+                var throttle = mapping.getLeft();
+                var reqs = mapping.getRight();
+                for (var req : reqs) {
+                    reqLists.computeIfAbsent(req.getLeft(), ignore -> new ArrayList<>())
+                            .add(Pair.of(throttle, req.getRight()));
+                }
+                newActiveThrottles.add(throttle);
+            } catch (IllegalStateException badBucket) {
+                log.error("When constructing bucket '{}' from state: {}", bucket.name(), badBucket.getMessage());
+            }
+        }
+        EnumMap<HederaFunctionality, ThrottleReqsManager> newFunctionReqs = new EnumMap<>(HederaFunctionality.class);
+        reqLists.forEach((function, reqs) -> newFunctionReqs.put(function, new ThrottleReqsManager(reqs)));
+
+        functionReqs = newFunctionReqs;
+        activeThrottles = newActiveThrottles;
+
+        logResolvedDefinitions(CAPACITY_SPLIT);
+    }
+
+    private ThrottleGroup<HederaFunctionality> hapiGroupFromPbj(
+            com.hedera.hapi.node.transaction.ThrottleGroup pbjThrottleGroup) {
+        return new ThrottleGroup<>(pbjThrottleGroup.milliOpsPerSec(), pbjThrottleGroup.operations());
+    }
+
+    private void logResolvedDefinitions(int capacitySplit) {
+        var sb = new StringBuilder(
+                "Resolved handle throttles (after splitting capacity " + capacitySplit + " ways) - \n");
+        functionReqs.entrySet().stream()
+                .sorted(Comparator.comparing(entry -> entry.getKey().toString()))
+                .forEach(entry -> {
+                    var function = entry.getKey();
+                    var manager = entry.getValue();
+                    sb.append("  ")
+                            .append(function)
+                            .append(": ")
+                            .append(manager.currentUsage()) // use current usage instead of the package private
+                            // asReadableRequirements(), otherwise we need to make it public as
+                            // well
+                            .append("\n");
+                });
+        log.info("{}", () -> sb.toString().trim());
+    }
+
+    public GasLimitDeterministicThrottle gasLimitThrottle() {
+        return gasThrottle;
+    }
+
+    //    @Override
+    //    public boolean shouldThrottleQuery(@NonNull HederaFunctionality functionality, @NonNull Query query) {
+    //        return false;
+    //    }
```

Comment:
Is this still to be implemented or did you just forget about that commented code?

Reply:
Good point, I missed to add a TODO here because I was not sure if we need to do query throttling in the backend throttle (I see there is an implementation in mono for this but did not find any actual usages).